src/SWP/Bundle/CoreBundle/EventSubscriber/RevisionsSubscriber.php

```diff
@@ -0,0 +1,125 @@
+
+    /**
+     * @var FactoryInterface
+     */
+    protected $revisionLogFactory;
+
+    /**
+     * RevisionsSubscriber constructor.
+     *
+     * @param ContainerRepository $containerRepository
+     * @param FactoryInterface    $revisionLogFactory
+     */
+    public function __construct(ContainerRepository $containerRepository, FactoryInterface $revisionLogFactory)
+    {
+        $this->containerRepository = $containerRepository;
+        $this->revisionLogFactory = $revisionLogFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::REVISION_PUBLISH => 'publish',
+        ];
+    }
+
+    /**
+     * @param RevisionPublishedEvent $event
+     */
+    public function publish(RevisionPublishedEvent $event)
+    {
+        /** @var RevisionInterface $revision */
+        $revision = $event->getRevision();
+        if (null === $revision->getPrevious()) {
+            return;
+        }
+
+        // new revision containers id's
+        $queryBuilder = $this->containerRepository->createQueryBuilder('c');
+        $newRevisionContainers = $queryBuilder->select('c.uuid')
+            ->where('c.revision = :revision')
+            ->setParameter('revision', $revision)
+            ->getQuery()
+            ->getResult();
+
+        $ids = [];
+        foreach ($newRevisionContainers as $container) {
+            $ids[] = $container['uuid'];
+        }
+
+        // published revisions containers
+        $criteria = new Criteria();
+        $criteria->set('revision', $revision->getPrevious());
+        $queryBuilder = $this->containerRepository->getQueryByCriteria($criteria, [], 'c');
+        if (count($ids) > 0) {
+            $queryBuilder->andWhere('c.uuid NOT IN (:ids)')->setParameter('ids', $ids);
+        }
+
+        $containers = $queryBuilder->getQuery()->getResult();
+        /** @var ContainerInterface|RevisionAwareInterface $container */
+        foreach ($containers as $container) {
+            $container->setRevision($revision);
+            $this->log($container, $revision);
+        }
+
+        $this->containerRepository->flush();
+    }
+
+    /**
+     * @param $object
+     * @param RevisionInterface $revision
+     */
+    private function log($object, $revision)
+    {
+        if (!$object instanceof PersistableInterface || !$object instanceof RevisionAwareInterface) {
+            return;
+        }
+
+        /** @var RevisionLogInterface|PersistableInterface $revisionLog */
+        $revisionLog = $this->revisionLogFactory->create();
+        $revisionLog->setEvent(RevisionLogInterface::EVENT_UPDATE);
+        $revisionLog->setObjectType(get_class($object));
+        $revisionLog->setObjectId($object->getId());
+        $revisionLog->setSourceRevision($revision->getPrevious());
+        $revisionLog->setTargetRevision($revision);
+
+        $this->containerRepository->persist($revisionLog);
```

Comment:
There is already `add` method and it's doing `persist` and `flush` at once. Adding those methods there just give us better control on process and don't injecting `object manager` (already injected into repository).